src/feature/control/control_cmd.c

```diff
@@ -1430,7 +1430,7 @@ handle_control_hsfetch(control_connection_t *conn,
              rend_valid_descriptor_id(arg1 + v2_str_len) &&
              base32_decode(digest, sizeof(digest), arg1 + v2_str_len,
                            REND_DESC_ID_V2_LEN_BASE32) ==
-                REND_DESC_ID_V2_LEN_BASE32) {
+                DIGEST_LEN) {
```

Comment:
Can we just use `sizeof(digest)` here so if it ever (even though impossible) change size, we caught it. Just good practice overall in tor.git.

Thanks!